Library/Homebrew/cask/lib/hbc/cask.rb

```diff
@@ -129,5 +129,32 @@ def dumpcask
         odebug "Cask instance method '#{method}':", send(method).to_yaml
       end
     end
+
+    def to_hash
+      {
+        "name" => name,
+        "homepage" => homepage,
+        "url" => url,
+        "appcast" => appcast,
+        "version" => version,
+        "sha256" => sha256,
+        "artifacts" => artifacts.map do |a|
+          if a.respond_to? :to_a
+            a.to_a
+          elsif a.methods.include? :to_h
```

Comment:
Also use `a.respond_to?` here. And check for `to_h` first, because a `Hash` also responds to `to_a`.